lib/Toc/ToctreeBuilder.php

```diff
@@ -42,13 +45,20 @@ public function buildToctreeFiles(
                 $globFiles = $this->globSearcher
                     ->globSearch($environment, $globPattern);
 
+                asort($globFiles);
+
                 foreach ($globFiles as $globFile) {
                     // if glob finds a file already explicitly defined
                     // don't duplicate it in the toctree again
                     if (in_array($globFile, $toctreeFiles, true)) {
                         continue;
                     }
 
+                    if ($globFile === $environment->absoluteUrl($environment->getCurrentFileName())) {
+                        // filter out the current file from being added as a glob
+                        continue;
+                    }
```

Comment:
This is a behavior change in this library. It only affects glob toctrees:

```
.. toctree::
    :glob:

    *
```

Thanks to this change, this will no longer include the CURRENT file when matching the `*`. That is how Sphinx behaves.